### DISCUSSION TITLE
bad optional access

### DISCUSSION THREAD

**Discussion by JaesukLim** [ANLAB-KAIST/KENSv3#182] [Project Q&A 2. Transport Layer (TCP)] [answered]
It seems there are a few question asking about this error.
I saw that error after sendPacket is called.
![image](https://github.com/ANLAB-KAIST/KENSv3/assets/143942994/273b1e04-a18d-4ac4-8ae3-bc533f8d2aef)
when I debugged, I found the that occurred after line 149 from E_System.cpp

Also, when I tried --gtest_catch_exceptions=0, it just said 
terminate called after throwing an instance of 'std::bad_optional_access'
  what():  bad optional access
Signal: SIGABRT (Aborted)

Is there any clue for this error?

**Comment by Pusnow** [comment]
What parameters do you use to call `sendPacket()`?

**Reply by JaesukLim** [reply]
I used 
sendPacket("IPv4", std::move(res_packet));

**Reply by JaesukLim** [reply]
After that, when I saw the pcap file, the packet is actually not sent

**Reply by Pusnow** [reply]
Hmm... Would you send your source code (TCPAssignment.cpp and hpp) to TAs? cs341.pj@kaist.ac.kr

**Reply by JaesukLim** [reply]
Please check the email! 
I appreciate for your quick answer!

**Reply by Pusnow** [reply, marked as answer]
It turns out that you have set the wrong IP address for sending packets.
Additionally, you parsed the packet incorrectly. Please, double-check your packet parsing code.

**Reply by JaesukLim** [reply]
Thank you for your help!
